modules/text/CMakeLists.txt

```diff
@@ -4,11 +4,42 @@ ocv_define_module(text opencv_ml opencv_imgproc opencv_core opencv_features2d op
 if(NOT CMAKE_CROSSCOMPILING OR OPENCV_FIND_TESSERACT)
   set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${CMAKE_CURRENT_SOURCE_DIR}/cmake)
   find_package(Tesseract QUIET)
+  find_package(Leptonica QUIET)
```

Comment:
Try this code:
```
if(NOT CMAKE_CROSSCOMPILING OR OPENCV_FIND_TESSERACT)
  find_package(Tesseract QUIET)  # Prefer CMake's standard locations (including Tesseract_DIR)
  if(NOT Tesseract_FOUND)
    include("${CMAKE_CURRENT_SOURCE_DIR}/cmake/FindTesseract.cmake")  # OpenCV's fallback
  endif()
  if(Tesseract_FOUND)
```
(other changes should be reverted)

You can use `ocv_cmake_dump_vars(tesseract)` for debugging of search results.

Reply:
About my build:
OS Windows 10, VS 2017, CMake 3.12.
To properly work Tesseract require that Leptonica must be built with image proc. libs. At least with tiff and libpng. In turn, libpng require zlib, therefore, except Tesseract and Leptonica we need to build it too.
I didnt use guide, just configured each lib through CMake GUI and build via msbuild.
Shared build opencv and libs was successful and work fine, but in statical build, we receive unresolved external errors to libs with which Leptonica was built.
In case that Leptonica was configured using find_package ${Leptonica_LIBRARIES} consist image proc. dependencies and ocv_target_link_libraries successfully added it to opencv_text module VS project.
Goal this patch first was resolve Windows static build via VS.
I have now took into account of your comments and make changes according to them.
This 
> Try this code:
> 
> if(NOT CMAKE_CROSSCOMPILING OR OPENCV_FIND_TESSERACT)
>   find_package(Tesseract QUIET)  # Prefer CMake's standard locations (including Tesseract_DIR)
>   if(NOT Tesseract_FOUND)
>     include("${CMAKE_CURRENT_SOURCE_DIR}/cmake/FindTesseract.cmake")  # OpenCV's fallback
>   endif()
>   if(Tesseract_FOUND)
code I not tested because it not fix Win static build.

After update I test this patch in Windows 10 and Xubuntu 18.04.1 LTS.
In Windows I try 2 options: 
- specify Tesseract_DIR and Leptonica_DIR (static build)
- specify Tesseract_INCLUDE_DIR, Tesseract_LIBRARY and Lept_LIBRARY (shared build).
Both of them works fine. Build was successful and example_text_end_to_end_recognition work correct.
In Xubuntu I install libtesseract-dev and it dependencies. CMake properly detect Tesseract_INCLUDE_DIR, Tesseract_LIBRARY and Lept_LIBRARY. Build also was successful and example_text_end_to_end_recognition work correct.